### PR TITLE
Add more documentation

### PR DESCRIPTION
**Pull request by brandonfoong**
Improve the following sections of the DG:
- Architecture
- Non-functional requirements
- Glossary
- Testing

Check https://brandonfoong.github.io/tp/DeveloperGuide.html for the rendered DG

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -151,37 +126,80 @@ Through Patient Manager, general practitioners are able to manage patients faste
...
-runtest.bat
-```
-
-### Manual Testing
-The current version does not support storing the data on the local drive. To start with the manual testing process, you may refer to the "QuickStart" section of the [User Guide](UserGuide.md).
+   be able to accomplish most of the tasks faster using commands than using the mouse.
+4. The data should be stored locally and should be in a human editable text file.
+5. The application should work without requiring an installer.
+6. The application should be at most 100 MB in size.
+7. The application should not rely on any remote server, or database management system.
```
Comment (fsgmhoward) [inline docs/DeveloperGuide.md:138]: As per SE-EDU's markdown guide, using all `1.` might be better.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -94,12 +66,14 @@ It is also responsible for displaying the response from PatientManager to the sc
 
 The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add S1234567D`.
 
-{TO BE ADDED}
+<img src="./images/sequence_diagram.png" width="600">
```
Comment (fsgmhoward) [inline docs/DeveloperGuide.md:69]: Would it be better if to use markdown's inbuilt tag instead?
Like this one for [our user guide](https://github.com/AY2021S2-CS2113T-W09-4/tp/blame/master/docs/UserGuide.md#L38)!

**Review by fsgmhoward — COMMENTED** [review]
Looks great 👍 Much appreciated.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (brandonfoong) [reply docs/DeveloperGuide.md:69]: I was considering that, but it seems that github flavoured markdown doesn't support resizing images from within the tag (see https://gist.github.com/uupaa/f77d2bcf4dc7a294d109). It seems like I have to fall back on `` tags to control the image size.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (brandonfoong) [reply docs/DeveloperGuide.md:138]: Done!

**Review by Emkay16 — APPROVED** [review]
Looks great! 👍

**Pull request merged**